### PR TITLE
feat: short lived transactions

### PR DESCRIPTION
**Pull request by mping-exo**
- Move a bunch of sql tests to its own namespace
- Ensure that processing leafs occurs outside a transaction, so workers don't hold the connection for too long :tada: 
- Allow marking all pending tasks as failed
- Add more tests to ensure retry works as expected

Retries can be resubmitted on eg: app start:
```
(let [workflow-ids (ablauf/fail-pending-tasks! db store)] 
  (doseq [workflow-id workflow-ids]
    (when 
      (ablauf/submit-retry db store workflow-id)
       (log/info "Workflow retried submitted successfully")))
```

### CODE REVIEW

File: resources/migrations/20231023083307-task-pending-status.down.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE task MODIFY COLUMN
```
Comment (mping-exo) [inline resources/migrations/20231023083307-task-pending-status.down.sql:1]: We will need to run this on root-api also

File: test/ablauf/job/sql_worker_fail_test.clj [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+(defn- mock-store []
+  (reify store/JobStore
+    (persist [this uuid context state]
+      (d/success-deferred :ok))))
+
+(defn- action-fn [{:ast/keys [action payload]}]
+  (case action
+    :action/fail (d/error-deferred :error/error)
+    :action/error (throw (Error. "This isn an error that is not caught"))
```
Comment (alexanderjamesking) [inline test/ablauf/job/sql_worker_fail_test.clj:23]:
```suggestion
    :action/error (throw (Error. "This is an error that is not caught"))
```

**Comment by alexanderjamesking** [comment]
I don't see a test that actually checks a task was set to pending, it could be useful to add a test that doesn't make use of the polling job but where we call `process-one` and verify what happens after each step.

**Comment by wdullaer** [comment]
If you know that only a single process will run tasks / modify the table at one point in time, you can do what we do in warp-jobs: at system startup we move all `pending` tasks to `failed`. These were running when we crashed, we can't resume them, so we have to assume they're failed.

This way you don't have to define a timeout, which will be extremely hard to get right since some leaves take fractions of second and others can run for hours.

**Comment by mping-exo** [comment]
That sounds a bit more sensible to me, I would prefer that all decisions are taken by the library consumers. Does it sound right to you @alexanderjamesking ?

We can eventually extend that behaviour to 1) mark as failed 2) retry if idempotent so we get "free" resume capability, but I would suggest doing that on another PR

**Comment by wdullaer** [comment]
If we do want a sharded runtime in the future, we can add an "owner" field. If we use something like statefulset, we'll get a fixed identity in k8s, and so we can then only fail leafs that were owned by ourselves at startup, again sidestepping the timer problem.

Maybe we spend 30 mins this afternoon together syncing our thoughts?

**Review by wdullaer — APPROVED** [review]

**Pull request merged**